unison-share-api/src/Unison/Server/Backend.hs

```diff
@@ -249,20 +252,42 @@ loadReferentType codebase = \case
           ++ show r
 
 data TermEntry v a = TermEntry
-  { termEntryReferent :: Referent,
-    termEntryName :: HQ'.HQSegment,
+  { termEntryReferent :: V2Referent.Referent,
+    termEntryHash :: ShortHash,
+    termEntryName :: NameSegment,
+    termEntryConflicted :: Bool,
     termEntryType :: Maybe (Type v a),
-    termEntryTag :: Maybe TermTag
+    termEntryTag :: TermTag
   }
   deriving (Eq, Ord, Show, Generic)
 
+termEntryDisplayName :: TermEntry v a -> Text
+termEntryDisplayName = HQ'.toText . termEntryHQName
+
+termEntryHQName :: TermEntry v a -> HQ'.HashQualified NameSegment
+termEntryHQName (TermEntry {termEntryName, termEntryConflicted, termEntryHash}) =
+  if termEntryConflicted
+    then HQ'.HashQualified termEntryName termEntryHash
+    else HQ'.NameOnly termEntryName
+
 data TypeEntry = TypeEntry
   { typeEntryReference :: Reference,
-    typeEntryName :: HQ'.HQSegment,
+    typeEntryHash :: ShortHash,
```

Comment:
What happens if we drop `ShortHash` here and rely on `typeEntryReference`?

Reply:
Later on we convert a `TypeEntry` into the API representation, which is where we'd want to construct the ShortHash. Moving the short hash construction to there means we need to pass in the Codebase and make the function and any function which calls it effectful; so I just passed the shorthash here to avoid 'dirtying' a bunch of pure code.

This wouldn't be an issue if we just decided on a single global shorthash length at the API level though; should we do that?